### DISCUSSION TITLE
Duplicating query. When changing, At Run Query query schema is not updated . Budibase 2.11.33

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#12134] [Help]
Duplicating query. When changing, At Run Query query schema is not updated. Budibase 2.11.33

Steps: Duplicated an existing query. Changed sql code and bindings. Executed query. At sidepanel scema is of old query. Needed to delete all schema fields to allow Budibase create correct schema. 
In general Handling of Custom queries gotten worse compared to version 2.8. New layout is confusing.

**Comment by ConorWebb96** [comment]
Hey @gmentzik,

Did you save after running your query? Once you have changed the query and re-ran you must save otherwise if you leave it will default back to the previous version. I believe this to be the same before the update. The only difference being is the save button has changed places, instead of being at the bottom or near the middle its right beside the `Run query` button. (Judging from what you've written I believe you missed the saving step before duplicating.)
<img width="938" alt="Screenshot 2023-10-23 at 08 19 43" src="https://github.com/Budibase/budibase/assets/126772285/dbcc588b-e013-4709-b473-8f44a9136a03">

I've raised this with the product team, additional amends to improve the query page's layout might happen once they have reviewed.

Thanks!
